Add next/previous at the end of extra pages

jonatanklosko opened the pull request:
An initial idea for #1295.

Currently this adds next/previous links at the end of every page from *extras*. I'm not sure if this makes much sense for modules/tasks, as those are rather not read linearly.

You can see the build preview [here](https://static.jonatanklosko.com/ex-doc/issue1295/initial/readme.html). Screenshots for the reference:

**Desktop**

![image](https://user-images.githubusercontent.com/17034772/101418064-491d6580-38ed-11eb-81f3-2508b8f7b2c2.png)

**Mobile**

![image](https://user-images.githubusercontent.com/17034772/101418081-55092780-38ed-11eb-8fc6-212bf6adb051.png)

josevalim commented:
This looks amazing! Thank you.

Just 3 comments:
1. I am afraid the tittle font for the links is too big. For example how would it look like in a project like [Ecto](https://hexdocs.pm/ecto/getting-started.html)?

2. I am wondering if the "continue reading" can be confusing when guides are not related to each other like in exdoc. Maybe only using "next" is enough?

3. We also need tests.

jonatanklosko commented:
> 1. I am afraid the tittle font for the links is too big. For example how would it look like in a project like Ecto?

Good point, I somehow assumed the titles are kept short, but Ecto is a perfect counterexample here.

> 2. I am wondering if the "continue reading" can be confusing when guides are not related to each other like in exdoc. Maybe only using "next" is enough?

Wasn't sure on that either, this is most suitable for guides, but using a more generic label sounds more reasonable.

If we kept the normal font size that's how it would look (also alternative captions):

![image](https://user-images.githubusercontent.com/17034772/101513795-de1b6f80-397c-11eb-9b9f-4cfc9b8a123f.png)

What do you think?

---

> 3. We also need tests.

Sure, just wanted to see if that's the right direction first. Is `html/templates_test.exs` the right place for those in this case?

jonatanklosko commented:
I've updated [the preview](https://static.jonatanklosko.com/ex-doc/issue1295/initial/readme.html) with those changes and added tests, let me know if you have any other thoughts!

josevalim commented:
Should the “next page” be text aligned to the right?

jonatanklosko commented:
For short names it looks a bit weird, but I don't have a strong opinion on that.

**Aligned right**

Long
![image](https://user-images.githubusercontent.com/17034772/101520347-a7495780-3984-11eb-9b32-3569644ab3a0.png)

Short
![image](https://user-images.githubusercontent.com/17034772/101520432-c8aa4380-3984-11eb-898c-5ac42c5a8c6b.png)

**Aligned left**

Long
![image](https://user-images.githubusercontent.com/17034772/101520370-ae706580-3984-11eb-889e-ac0a2ce76853.png)

Short
![image](https://user-images.githubusercontent.com/17034772/101520414-c21bcc00-3984-11eb-978a-11fa54bfd99b.png)

josevalim commented:
Or the “previous page” can be aligned to the right so both “next page” and “previous page” are side by side in the center?

jonatanklosko commented:
And here's the "previous page" aligned right. I like this because the alignment is clearly visible, because the text sticks to the edge rather than the arrow.

Long
![image](https://user-images.githubusercontent.com/17034772/101520881-5c7c0f80-3985-11eb-8649-c1da6a975a51.png)

Short
![image](https://user-images.githubusercontent.com/17034772/101520786-3eaeaa80-3985-11eb-9ee3-ac4599eced8d.png)

josevalim commented:
Awesome, let’s go with left aligned right then! Once we have tests, we can probably ship it! :D

jonatanklosko commented:
I've updated the alignment. Do the tests look ok?

josevalim commented on test/ex_doc/formatter/html/templates_test.exs:
```diff
@@ -463,4 +463,33 @@ defmodule ExDoc.Formatter.HTML.TemplatesTest do
     end
   end
+
+  describe "bottom actions" do
```
I was hoping more of an integration test, perhaps in html_test, as this one does not really assert we have built the refs properly (which is the complex part!). Do we have two guides in our test suite? It should be enough that generating docs for our existing guides and match on the output.

jonatanklosko replied:
Sure, should've done that before!

josevalim commented:
Absolutely fantastic, thank you! :green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:
